web/src/util/serialization/activities.js

```diff
@@ -0,0 +1,239 @@
+import { applyToNumbers, generateKey } from '../index';
+
+/**
+ * Serializes an APD object from redux state shape into
+ * API format
+ * @param {Object} activityState Activity object from redux state
+ */
+export const toAPI = activityState => {
+  const activity = {
+    id: activityState.id,
+    name: activityState.name,
+    fundingSource: activityState.fundingSource,
+    summary: activityState.descShort,
+    description: activityState.descLong,
+    alternatives: activityState.altApproach,
+    costAllocationNarrative: {
+      methodology: activityState.costAllocationDesc,
+      otherSources: activityState.otherFundingDesc
+    },
+    costAllocation: Object.entries(activityState.costAllocation).map(
+      ([year, allocation]) => ({
+        federalPercent: +allocation.ffp.federal / 100,
+        statePercent: +allocation.ffp.state / 100,
+        otherAmount: +allocation.other,
+        year
+      })
+    ),
+    goals: activityState.goals.map(g => ({
+      id: g.id,
+      description: g.desc,
+      objective: g.obj
+    })),
+    schedule: activityState.milestones.map(m => ({
+      id: m.id,
+      milestone: m.name,
+      plannedStart: m.start || undefined,
+      plannedEnd: m.end || undefined
+    })),
+    statePersonnel: activityState.statePersonnel.map(s => ({
+      id: s.id,
+      title: s.title,
+      description: s.desc,
+      years: Object.keys(s.years).map(year => ({
+        cost: +s.years[year].amt,
+        fte: +s.years[year].perc / 100,
+        year
+      }))
+    })),
+    contractorResources: activityState.contractorResources.map(c => ({
+      id: c.id,
+      name: c.name,
+      description: c.desc,
+      start: c.start || undefined,
+      end: c.end || undefined,
+      years: Object.keys(c.years).map(year => ({
+        cost: +c.years[year],
+        year
+      }))
+    })),
+    expenses: activityState.expenses.map(e => ({
+      id: e.id,
+      category: e.category,
+      description: e.desc,
+      entries: Object.keys(e.years).map(year => ({
+        amount: +e.years[year],
+        year
+      }))
+    })),
+    standardsAndConditions: {
+      businessResults: activityState.standardsAndConditions.bizResults,
```

Comment:
at some point, we should unify all of these names (i.e., businessResults v. bizResults), would be greatly simplify this object